scylla/tests/integration/session.rs

```diff
@@ -3256,16 +3279,253 @@ async fn test_vector_type_prepared() {
         .unwrap();
 
     let prepared_statement = session
+        .prepare(format!("INSERT INTO {}.t (a, b, c) VALUES (?, ?, ?)", ks))
+        .await
+        .unwrap();
+    session
+        .execute_unpaged(
+            &prepared_statement,
+            &(2, vec![11, 12, 13, 14], vec!["afoo", "abar"]),
+        )
+        .await
+        .unwrap();
+
+    let query_result = session
+        .query_unpaged(format!("SELECT * FROM {}.t", ks), &[])
+        .await
+        .unwrap();
+
+    let rows: Vec<(i32, Vec<i32>, Vec<String>)> = query_result
+        .into_rows_result()
+        .unwrap()
+        .rows::<(i32, Vec<i32>, Vec<String>)>()
+        .unwrap()
+        .map(|r| r.unwrap())
+        .collect();
+    assert_eq!(
+        rows[0],
+        (
+            2,
+            vec![11, 12, 13, 14],
+            vec!["afoo".to_string(), "abar".to_string()]
+        )
+    );
+}
+
+async fn test_vector_single_type<
+    T: SerializeValue + for<'a> DeserializeValue<'a, 'a> + PartialEq + Debug + Clone,
+>(
+    keyspace: &str,
+    session: &Session,
+    type_name: &str,
+    values: Vec<T>,
+) {
+    let table_name = format!(
+        "test_vector_{}",
+        type_name
+            .replace("<", "A")
+            .replace(">", "B")
+            .replace(",", "C")
+    );
+    let query = format!(
+        "CREATE TABLE {}.{} (a int PRIMARY KEY, b vector<{}, {}>)",
+        keyspace,
+        table_name,
+        type_name,
+        values.len()
+    );
+    session.ddl(query).await.unwrap();
+
+    let prepared_insert = session
         .prepare(format!(
-            "INSERT INTO {}.t (a, b, c) VALUES (?, [11, 12, 13, 14], ['afoo', 'abar'])",
-            ks
+            "INSERT INTO {}.{} (a, b) VALUES (?, ?)",
+            keyspace, table_name
         ))
         .await
         .unwrap();
+    let prepred_select = session
+        .prepare(format!("SELECT * FROM {}.{}", keyspace, table_name))
+        .await
+        .unwrap();
     session
-        .execute_unpaged(&prepared_statement, &(2,))
+        .execute_unpaged(&prepared_insert, &(1, &values))
         .await
         .unwrap();
 
-    // TODO: Implement and test SELECT statements and bind values (`?`)
+    let query_result = session.execute_unpaged(&prepred_select, &[]).await.unwrap();
+
+    let result = query_result.into_rows_result().unwrap();
+
+    let rows: Vec<(i32, Vec<T>)> = result
+        .rows::<(i32, Vec<T>)>()
+        .unwrap()
+        .map(|r| r.unwrap())
+        .collect();
+
```

Comment:
I think there is a method like `single_row` which you can use instead of `rows` here to simplify code.